### ISSUE TITLE
Long inlay hints combined with soft wrap sometimes block moving up

### ISSUE THREAD

**Issue by AverageLinuxEnjoyer** [helix-editor/helix#9218]
### Summary

The k key (to move up) sometimes does nothing when I'm encountering an inlay hint that spans over multiple lines.

Here's the video. In the first half moving down and up works. In the second half (after I open the second terminal), moving down always works but moving up sometimes doesn't. When I highlight a line it means I'm blocked.

https://github.com/helix-editor/helix/assets/91616580/99eb74ae-c6cf-4e7b-b060-1aee72d12bb3

### Reproduction Steps

 

I haven't been able to easily get long types that reproduce the problem so instead of creating a minimal example I'll just give you my project, which is small anyway (1 code file).

1. have rust-analyzer or nix installed
2. clone this repo https://github.com/AverageLinuxEnjoyer/helix-bug-repo (contains the code in the video)
3. `nix develop -c $SHELL` if you have nix, but not rust-analyzer
4. `cargo build` to get `chumsky` (the crate in Cargo.toml) installed
5. `hx` into the project
6. enable inlay hints and soft wrap if not enabled
```toml
[editor.lsp]
display-inlay-hints = true

[editor.soft-wrap]
enable=true
```
7. try scrolling up and down in `main.rs` like I did with various terminal sizes

I expected this to happen:
To be able to move up unconditionally.

Instead, this happened:
I'm sometimes blocked and can't move up.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
There's no change in logs for this bug.
```

</details>

### Platform

Linux

### Terminal Emulator

kitty 0.28.1

### Installation Method

nixpkgs

### Helix Version

helix 23.05

**Comment by kirawi** [comment]
I don't think this was reported before so it's like still an issue, but I'd recommend upgrading to 23.10 anyway.

**Comment by ivq** [comment]
I'm facing the same issue on latest version 2d15acdf6029fd9b418bf4797c2947d6004dddbd.

**Comment by kirawi** [comment]
The relevant function is defined in https://github.com/helix-editor/helix/blob/2f4572d91e5846f2b300c9292c5b446474c5fe52/helix-core/src/movement.rs#L55-L104

The code to resolve this issue might also resolve `// TODO how to handle inline annotations that span an entire visual line (very unlikely).`

**Comment by pascalkuthe** [comment]
Like I said this will be addressed in #6417. These kinds of changes to the positioning system are highly interrelated

**Comment by ivq** [comment]
Hey, I can confirm that the bug no more exist in the latest version(25.01.1 as of now).
